projects/plugins/jetpack/modules/masterbar/featured-post-thumbnail/class-featured-post-thumbnail.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Featured Post Thumbnail Support on WP.com.
+ *
+ * Adds support for `post-thumbnail` image on get media endpoint which is used by Gutenberg featured image.
+ * This is only interim fix and should removed as soon as permanent fix is available.
+ * For more context see: https://github.com/Automattic/wp-calypso/issues/52010
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+/**
+ * Class Featured_Post_Thumbnail
+ */
+class Featured_Post_Thumbnail {
+
+	/**
+	 * FeaturedPostThumbnail constructor.
+	 * Register targeted filters
+	 */
+	public function __construct() {
+		add_filter( 'rest_prepare_attachment', array( $this, 'force_add_featured_post_thumbnail_image_support' ), 20, 2 );
```

Comment:
`rest_prepare_attachment` filter seems generic. Is there another more specific "Editor" filter? Can we hook only to the specific endpoint?

Reply:
The `rest_prepare_attachment` is only used by [wp-rest-attachments-controller.php](https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php#L832). AFAIK there is no other way to alter response of attachment endpoint.